epix/run_epix.py

```diff
@@ -0,0 +1,202 @@
+                                               density=epix.awkward_to_flat_numpy(result['xe_density']))
+    result['photons'] = epix.reshape_awkward(photons, ak.num(result['ed']))
+    result['electrons'] = epix.reshape_awkward(electrons, ak.num(result['ed']))
+    if args['debug']:
+        _ = monitor_time(tnow, 'get quanta.')
+
+    # Separate event in time
+    number_of_events = len(result["t"])
+    if args['source_rate'] == -1:
+        # Only needed for a clean separation:
+        result['t'] = result['t'] - result['t'][:, 0]
+        result = result[result['t'] <= args['max_delay']]
+
+        dt = epix.times_for_clean_separation(number_of_events, args['max_delay'])
+        if args['debug']:
+            print('Clean event separation')
+    elif args['source_rate'] == 0:
+        # In case no delay should be applied we just add zeros
+        dt = np.zeros(number_of_events)
+    else:
+        # Rate offset computed based on the specified rate and job_id.
+        # Assumes all jobs were started with the same number of events.
+        offset = (args['job_id']*n_simulated_events)/args['source_rate']
+        dt = epix.times_from_fixed_rate(args['source_rate'],
+                                        number_of_events,
+                                        n_simulated_events,
+                                        offset
+                                        )
+        if args['debug']:
+            print(f"Fixed event rate of {args['source_rate']} Hz")
+
+    result['t'] = result['t'][:, :] + dt
+
+    # Reshape instructions:
+    instructions = epix.awkward_to_wfsim_row_style(result)
+    if not args['source_rate'] == 0:
+        # Only sort by time if source rates are applied.
+        instructions = np.sort(instructions, order='time')
+
+    ins_df = pd.DataFrame(instructions)
+
+    if return_df:
+        if args['output_path'] and not os.path.isdir(args['output_path']):
+            os.makedirs(args['output_path'])
+
+        output_path_and_name = os.path.join(args['output_path'], args['file_name'][:-5] + "_wfsim_instructions.csv")
+        if os.path.isfile(output_path_and_name):
+            warnings.warn("Output file already exists - Overwriting")
+        ins_df.to_csv(output_path_and_name, index=False)
+
+        print('Done')
+        print('Instructions saved to ', output_path_and_name)
+        if args['debug']:
+            _ = monitor_time(starttime, 'run epix.')
+
+    if return_wfsim_instructions:
+        return instructions
+
+
+def monitor_time(prev_time, task):
+    t = time.time()
+    print(f'It took {(t - prev_time):.4f} sec to {task}')
+    return t
+
+
+def setup(args):
+    """
+    Function which sets-up configurations. (for strax conversion)
+    Is returning the dict like this nessecairy?
+    :return:
+    """
+    # Getting file path and split it into directory and file name:
+    if not ('path' in args and 'file_name' in args):
+        p = args['input_file']
+        p = p.split('/')
```

Comment:
I think this fails if the file is in the same folder where epix is called and only the input file name is given.

Reply:
I have not added it, so I nether checked.